src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapper.java

```diff
@@ -325,7 +304,7 @@ public KNNVectorFieldMapper build(BuilderContext context) {
                 copyToBuilder,
                 ignoreMalformed,
                 stored.getValue(),
-                hasDocValues.getValue(),
+                hasDocValues.isConfigured() ? hasDocValues.getValue() : false,
```

Comment:
Not changing the default but adding in a check for explicit value setting on the mapping. 
@jmazanec15 WDYT?

Reply:
I would say just set it to false - is there a bwc concern?

Reply:
I was chewing a bit on that - thinking of BwC in general. From the code flow, it does not look like LuceneFieldMapper and MethodFieldMapper are serialized directly.
We use fromXContent and toXContent - which specifies all the initialized values.
I dont think we will break BwC with it.

Changing the default would entail changing defaults for FlatVectorField and ModelFieldMapper as well.

With this approach - I only change it for the general engine mappers.

Reply:
Updated the default to false except for flat fields.